packages/gestalt/src/Tabs.js

```diff
@@ -73,45 +128,137 @@ function Tab({
           rounding="pill"
           userSelect="none"
         >
-          <Text color={isActive ? 'white' : 'darkGray'} weight="bold" overflow="noWrap">
-            {children}
-          </Text>
-          {indicator === 'dot' && <Circle />}
+          <Flex alignItems="center" gap={2} justifyContent="center">
+            <Text color={isActive ? 'white' : 'darkGray'} weight="bold" overflow="noWrap">
+              {text}
+            </Text>
+
+            {indicator === 'dot' && <Dot />}
+          </Flex>
         </Box>
       </Link>
     </Box>
   );
 }
 
+const TAB_ROUNDING = 2;
+const TAB_INNER_PADDING = 1;
+
+function TabV2({
+  href,
+  indicator,
+  id,
+  index,
+  isActive,
+  onChange,
+  // No longer supported, will be removed when this variant ships
+  // eslint-disable-next-line no-unused-vars
+  size,
+  text,
+}: {|
+  ...TabType,
+  index: number,
+  isActive: boolean,
+  onChange: OnChangeHandler,
+  size: 'md' | 'lg',
+|}) {
+  const [hovered, setHovered] = useState(false);
+  const [focused, setFocused] = useState(false);
+  const [pressed, setPressed] = useState(false);
+
+  let color = 'white';
+  if (pressed) {
+    color = 'lightWash';
+  } else if ((hovered || focused) && !isActive) {
+    color = 'lightGray';
+  }
+
+  return (
+    <Box id={id} padding={3}>
+      <TapArea
+        href={href}
+        onBlur={() => setFocused(false)}
+        onFocus={() => setFocused(true)}
+        onMouseDown={() => setPressed(true)}
+        onMouseUp={() => setPressed(false)}
+        onMouseEnter={() => setHovered(true)}
+        onMouseLeave={() => setHovered(false)}
+        onTap={({ event }) => onChange({ activeTabIndex: index, event })}
+        role="link"
+        rounding={TAB_ROUNDING}
+        tapStyle="compress"
+      >
+        <Flex alignItems="center" direction="column">
+          <Box
+            color={color}
+            padding={TAB_INNER_PADDING}
+            position="relative"
+            rounding={TAB_ROUNDING}
+            userSelect="none"
+          >
+            <Flex alignItems="center" gap={2} justifyContent="center">
+              <Text color="darkGray" overflow="noWrap" weight="bold">
+                {text}
+              </Text>
+
+              {indicator === 'dot' && <Dot />}
+              {/* Flow is dumb and doesn't realize Number.isFinite will return false for a string or undefined */}
+              {typeof indicator === 'number' && Number.isFinite(indicator) && (
```

Comment:
`typeof` here is very 🙄

Reply:
@dangerismycat does it work with `Number.isInteger()`? Or do we also need to support floats?

Reply:
It doesn't unfortunately. We don't need to support floats, `.isFinite` is just my go-to for determining if a value is a number (and not NaN). I can update to `.isInteger` if you think that's the better choice.